csds-client/client/util.go

```diff
@@ -0,0 +1,259 @@
+				}
+			}
+		}
+	}
+	return ""
+}
+
+// TypeResolver implements protoregistry.ExtensionTypeResolver and protoregistry.MessageTypeResolver to resolve google.protobuf.Any types
+type TypeResolver struct{}
+
+func (r *TypeResolver) FindMessageByName(message protoreflect.FullName) (protoreflect.MessageType, error) {
+	return nil, protoregistry.NotFound
+}
+
+// FindMessageByURL links the message type url to the specific message type
+// TODO: If there's other message type can be passed in google.protobuf.Any, the typeUrl and messageType need to be added to this method to make sure it can be parsed and output correctly
+func (r *TypeResolver) FindMessageByURL(url string) (protoreflect.MessageType, error) {
+	switch url {
+	case "type.googleapis.com/envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager":
+		httpConnectionManager := envoy_config_filter_network_http_connection_manager_v2.HttpConnectionManager{}
+		return httpConnectionManager.ProtoReflect().Type(), nil
+	case "type.googleapis.com/envoy.api.v2.Cluster":
+		cluster := envoy_api_v2.Cluster{}
+		return cluster.ProtoReflect().Type(), nil
+	case "type.googleapis.com/envoy.api.v2.Listener":
+		listener := envoy_api_v2.Listener{}
+		return listener.ProtoReflect().Type(), nil
+	case "type.googleapis.com/envoy.config.filter.http.router.v2.Router":
+		router := envoy_config_filter_http_router_v2.Router{}
+		return router.ProtoReflect().Type(), nil
+	case "type.googleapis.com/envoy.api.v2.RouteConfiguration":
+		routeConfiguration := envoy_api_v2.RouteConfiguration{}
+		return routeConfiguration.ProtoReflect().Type(), nil
+	default:
+		return nil, protoregistry.NotFound
+	}
+}
+
+func (r *TypeResolver) FindExtensionByName(field protoreflect.FullName) (protoreflect.ExtensionType, error) {
+	return nil, protoregistry.NotFound
+}
+
+func (r *TypeResolver) FindExtensionByNumber(message protoreflect.FullName, field protoreflect.FieldNumber) (protoreflect.ExtensionType, error) {
+	return nil, protoregistry.NotFound
+}
+
+// parseConfigStatus parses each xds config status to string
+func parseConfigStatus(xdsConfig []*envoy_service_status_v2.PerXdsConfig) []string {
+	var configStatus []string
+	for _, perXdsConfig := range xdsConfig {
+		status := perXdsConfig.GetStatus().String()
+		var xds string
+		if perXdsConfig.GetClusterConfig() != nil {
+			xds = "CDS"
+		} else if perXdsConfig.GetListenerConfig() != nil {
+			xds = "LDS"
+		} else if perXdsConfig.GetRouteConfig() != nil {
+			xds = "RDS"
+		} else if perXdsConfig.GetScopedRouteConfig() != nil {
+			xds = "SRDS"
+		}
+		if status != "" && xds != "" {
+			configStatus = append(configStatus, xds+"   "+status)
+		}
+	}
+	return configStatus
+}
+
+// printOutResponse posts process response and print
+func printOutResponse(response *envoy_service_status_v2.ClientStatusResponse, fileName string) error {
+	fmt.Printf("%-50s %-30s %-30s \n", "Client ID", "xDS stream type", "Config Status")
```

Comment:
we should handle the case where the response is empty. We can just printout "No xDS clients connected."

Reply:
Added a check at the beginning of `printOutResponse()`: 
`if response.GetConfig() == nil || len(response.GetConfig()) == 0 {`, then printout "No xDS clients connected." and return